requirements.txt

```diff
@@ -3,14 +3,11 @@
 
 # dependencies
 numpy
-flake8==3.8.3
-pylint==2.5.3
+flake8==6.1.0
 pytest==8.0.0
-pytest-xdist==1.34.0
 
 # Pin importlib-metadata to <5 due to https://github.com/python/importlib_metadata/issues/409.
 importlib-metadata==4.13.0
```

Comment:
importlib-metadata was pinned because of an incompatibility with a certain version of flake8. If you are updating flake8 you may be able to remove this restriction on importlib-metadata and it wouldn't have to be in this file as it isn't a direct dependency.

Also, since almost everything in this file is pinned, you could consider listing unrestricted modules (numpy, flake8, pytest) in a requirements.in file and have pip-tools generate a fully pinned requirements.txt - we've been doing this in other projects and it makes everything more robust to external incompatibilities.